### PR TITLE
add MarketType to template creation

### LINKED ISSUE #3612: Market Creation Templates - Market Type Selections
- Create "Choose a market type" page
- Designs: https://www.figma.com/file/IpIAYVQPeQfuT5TTUboQpo4S/Market-creation?node-id=445%3A34890

image

### PR DESCRIPTION
**Pull request by andrewdamelio**
closes #3612

### CODE REVIEW

File: packages/augur-ui/src/modules/common/form.tsx [hunk 1]
```diff
@@ -509,10 +510,20 @@ export class CategorySingleSelect extends Component<
 > {
   state: CategorySingleSelectState = {
     selected: this.props.initialSelected || '',
-    value: this.props.initialValue || this.props.initialSelected || '',
+    value: this.props.initialValue || (this.props.initialSelected === CUSTOM ? '' : this.props.initialSelected || ''),
     showText: this.props.initialSelected === CUSTOM,
   };
 
+  componentWillReceiveProps(nextProps) {
```
Comment (bthaile) [inline packages/augur-ui/src/modules/common/form.tsx]: we are starting to get away from legacy react methods, use `componentDidUpdate`

**Review by bthaile — CHANGES_REQUESTED** [review]
one request for change, others look good. I see the PR is WIP, wanted to add feedback.

File: packages/augur-ui/src/modules/common/form.tsx (hunk 1 above)
Comment (andrewdamelio) [reply packages/augur-ui/src/modules/common/form.tsx]: ohhhhh interesting 👍

**Review by ghost — COMMENTED** [review]
This pull request has been automatically deployed to [FeaturePeek](https://featurepeek.com). 📈

 This link will always point to this PR's latest changes: https://peek.run/kq3hm9o

**Review by bthaile — APPROVED** [review]
Lgtm

**Comment by bthaile** [comment]
Well need to use sdk constants for market type. John can make the updates since he needs to use the sdk constants.

**Pull request merged**